### PR TITLE
Botprise integration with datadog

### PR DESCRIPTION
**Pull request by codechefnisha-bp**
### What does this PR do?
This PR shall list Botprise as one of the integrated customer

### Motivation

What inspired you to submit this pull request?

We have been using datadog Rest APIs to get usage metrics from devices. As part of integration we have installed datadog agent on few lab devices. With the help of agent we retrieve usage data from the devices and raise alert if usage cross threshold value. We have setup few monitors to check alerts for each host. For every alert we have been creating Service Now ticket which comes in botprise system with webhook integration.
Later while resolving those tickets we use datadog rest API to get usage metrics and host details.

### Review checklist
- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes
No, Please let me know if any required information would be missing.

### CODE REVIEW

File: botprise/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Agent Check: botprise
+
+## Overview
+
+We have been using datadog Rest APIs to get usage metrics from devices. As part of integration we have installed datadog agent on few lab devices. With the help of agent we retrieve usage data from the devices and raise alert if usage cross threshold value. We have setup few monitors to check alerts for each host. For every alert we have been creating Service Now ticket which comes in botprise system with webhook integration.
+Later while resolving those tickets we use datadog rest API to get usage metrics and host details. Also APIs are being used for querying process details
```
Comment (hithwen) [inline botprise/README.md:6]: This should be an overview about what the integration is about not a backstory ;). It should include a link to the service official page

File: botprise/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Setup
+
+### Installation
+
+The botprise check is included in the [Datadog Agent][2] package.
+No additional installation is needed on your server.
```
Comment (hithwen) [inline botprise/README.md:13]: This is not true for integrations-extras, please copy this section from other integrations in this repo

File: botprise/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+### Configuration
+1. Installed datadog agent on lab devices
+2. After successful installation devices got listed on datadog host list
+3. Created monitor for each of the host
+4. Configured monitor for metric and respective threshold value. Once threshold value would cross for the metric alert will be generated.
+5. Modified configuration to create ServiceNow ticket for each of the incoming alert
+6. Generated API key and Application key to trigger Rest API
```
Comment (hithwen) [inline botprise/README.md:21]: Please state instructions in imperative format

File: botprise/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+### Validation
+
+<Steps to validate integration is functioning as expected>
```
Comment (hithwen) [inline botprise/README.md:25]: Something is missing here?

File: botprise/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+## Data Collected
+
+### Metrics
+Botprise uses datadog metric API to get usage metrics from added devices. Also APIs are being used for querying process/infrastructure details
+
+For getting datadog API invoke we need to enter API key and application key
+
+Input a Datadog API Key []:xxxx9232ad913d1a864828a2df15xxxx
+Input a Datadog Application Key []:xxxxcb1798718f7a2da141071e7305599d60xxxx
```
Comment (hithwen) [inline botprise/README.md:35]: It looks this should be in the configuration section

File: botprise/README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+6. Generated API key and Application key to trigger Rest API
+
+### Validation
+
+<Steps to validate integration is functioning as expected>
+
+## Data Collected
+
+### Metrics
+Botprise uses datadog metric API to get usage metrics from added devices. Also APIs are being used for querying process/infrastructure details
```
Comment (hithwen) [inline botprise/README.md:30]: Here you should explain the metrics that are collected or refer to `metadata.csv`, please see other integrations as example

File: botprise/manifest.json [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  "display_name": "botprise",
+  "maintainer": "help@datadoghq.com",
+  "manifest_version": "1.0.0",
+  "name": "botprise",
+  "metric_prefix": "botprise.",
+  "metric_to_check": "",
+  "creates_events": false,
+  "short_description": "Integration with botprise to accelerate/optimize application performance",
+  "guid": "bbc51521-f87c-44c1-ba57-9c4e5dc23214",
+  "support": "core",
```
Comment (hithwen) [inline botprise/manifest.json:11]:
```suggestion
  "support": "contrib",
```

File: botprise/manifest.json [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  "supported_os": [
+    "linux",
+    "mac_os"
+  ],
+  "public_title": "Datadog-botprise integtration",
+  "categories": [
+    ""
+  ],
+  "type": "check",
+  "is_public": false,
```
Comment (hithwen) [inline botprise/manifest.json:21]:
```suggestion
  "is_public": true,
```

File: botprise/metadata.csv [hunk 9]
```diff
@@ -0,0 +1,4 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name
+system.cpu.user,gauge,,,,The percent of time the CPU spent running user space processes.,,system,
+system.cpu.system,,,,,,,,
+system.cpu.i,,,,,,,,
```
Comment (hithwen) [inline botprise/metadata.csv:4]: metrics should have a `botoprise.` prefix, type of metric, integration, description and short name are required fields

Comment (codechefnisha-bp) [reply botprise/metadata.csv]: @hithwen Thanks for reviewing the changes. Regarding metrics we have been using predefined metric to collect metrics from added server/client. Do we still need to populate values for metrics?

**Comment by codechefnisha-bp** [comment]
@hithwen I have incorporated review comments. Also we have setup email event push in datadog system. Botprise sends event to datadog whenever a new ticket comes in our system.

https://app.datadoghq.com/event/stream?tags_execution=and&show_private=true&per_page=30&aggregate_up=true&use_date_happened=false&display_timeline=true&from_ts=1596532140000&priority=normal&live=true&is_zoomed=false&status=all&to_ts=1596535740000&is_auto=false&incident=true&only_discussed=false&no_user=false&page=0&bucket_size=60000

Please let me know if I would be missing any thing.

File: botprise/README.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
+# Agent Check: botprise
```
Comment (apigirl) [inline botprise/README.md:1]:
```suggestion
```
this is pulled from the manifest

File: botprise/README.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+## Overview
+
+Botprise's Datadog integration allows you to send Botprise events to the event stream in Datadog. It submits Datadog events into the system over email.
```
Comment (apigirl) [inline botprise/README.md:5]: a link out to Botprise would be good

File: botprise/README.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+
+## Setup
+
+
+### Installation
+
+
+### Configuration
+1. Install datadog agent on lab devices
```
Comment (apigirl) [inline botprise/README.md:14]:
```suggestion
1. Install the Datadog Agent on your lab devices.
```

File: botprise/README.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+2. After successful installation devices will get listed on datadog host list(https://app.datadoghq.com/infrastructure/map)
```
Comment (apigirl) [inline botprise/README.md:15]:
```suggestion
2. After successful installation, your devices start sending data to Datadog. View the data on the [Datadog host list][3].
```

File: botprise/README.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+
+Input a Datadog API Key []:xxxx9232ad913d1a864828a2df15xxxx
+Input a Datadog Application Key []:xxxxcb1798718f7a2da141071e7305599d60xxxx
+
+## Troubleshooting
+
+Need help? Contact [Datadog support][1].
+
+[1]: https://docs.datadoghq.com/help/
+[2]: https://app.datadoghq.com/account/settings#agent
```
Comment (apigirl) [inline botprise/README.md:48]:
```suggestion
[2]: https://app.datadoghq.com/account/settings#agent
[3] https://app.datadoghq.com/infrastructure/map
[4]: https://docs.datadoghq.com/metrics/
[5]: https://docs.datadoghq.com/account_management/api-app-keys/
```

File: botprise/README.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+## Setup
+
+
+### Installation
+
+
+### Configuration
+1. Install datadog agent on lab devices
+2. After successful installation devices will get listed on datadog host list(https://app.datadoghq.com/infrastructure/map)
+3. Create monitor for each of the host
```
Comment (apigirl) [inline botprise/README.md:16]:
```suggestion
3. In Datadog, create a monitor for each of the hosts. Datadog generates alerts based on the monitor rules.
```

File: botprise/README.md [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+4. Configure monitor for metric and respective threshold value.
```
Comment (apigirl) [inline botprise/README.md:17]:
```suggestion
4. Configure each monitor for [metrics][4] and the respective threshold value. 
```

File: botprise/README.md [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+
+### Installation
+
+
+### Configuration
+1. Install datadog agent on lab devices
+2. After successful installation devices will get listed on datadog host list(https://app.datadoghq.com/infrastructure/map)
+3. Create monitor for each of the host
+4. Configure monitor for metric and respective threshold value. 
+5. Datadog generates alert based on the monitor rules.
```
Comment (apigirl) [inline botprise/README.md:18]:
```suggestion
```

File: botprise/README.md [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+6. Modify monitor configuration to create ServiceNow ticket for each of the incoming alert
```
Comment (apigirl) [inline botprise/README.md:19]:
```suggestion
5. Modify the monitor configuration to create a ServiceNow ticket for each of the incoming alerts.
```
link for ServiceNow?

File: botprise/README.md [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+7. Generate API key and Application key to call Datadog Rest APIs
```
Comment (apigirl) [inline botprise/README.md:20]:
```suggestion
6. Generate an [API key and an Application key][5] to call Datadog Rest APIs.
```

File: botprise/README.md [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+### Service Checks
+
+The Botprise integration does not include any service checks.
+
+### Events
+
+All events are sent to the Datadog event stream.
+
+### Configuration
+For getting datadog API invoke we need to enter API key and application key
```
Comment (apigirl) [inline botprise/README.md:38]:
```suggestion

To use the Datadog API, you need to enter an [API key and an application key][5]:
```

**Comment by codechefnisha-bp** [comment]
@apigirl Thanks for reviewing the changes. I have added suggested changes. Please approve.

File: botprise/README.md (hunk 20 above)
Comment (codechefnisha-bp) [reply botprise/README.md]: Added

File: botprise/README.md (hunk 19 above)
Comment (codechefnisha-bp) [reply botprise/README.md]: Done

File: botprise/README.md (hunk 18 above)
Comment (codechefnisha-bp) [reply botprise/README.md]: Added

File: botprise/README.md (hunk 17 above)
Comment (codechefnisha-bp) [reply botprise/README.md]: Removed

File: botprise/README.md (hunk 16 above)
Comment (codechefnisha-bp) [reply botprise/README.md]: Updated

File: botprise/README.md (hunk 15 above)
Comment (codechefnisha-bp) [reply botprise/README.md]: Corrected

File: botprise/README.md (hunk 14 above)
Comment (codechefnisha-bp) [reply botprise/README.md:45]: Updated

File: botprise/README.md (hunk 13 above)
Comment (codechefnisha-bp) [reply botprise/README.md]: Modified

File: botprise/README.md (hunk 11 above)
Comment (codechefnisha-bp) [reply botprise/README.md]: Updated

File: botprise/README.md (hunk 10 above)
Comment (codechefnisha-bp) [reply botprise/README.md]: Got it.Removed

**Comment by hithwen** [comment]
Hallo, you need to ensure that validation is passing, at the moment is failing because you need to add an entry to the `.github/CODEOWNERS` file

**Comment by codechefnisha-bp** [comment]
> Hallo, you need to ensure that validation is passing, at the moment is failing because you need to add an entry to the `.github/CODEOWNERS` file

@hithwen Thanks..All validations are passed now. Please approve.

**Comment by hithwen** [comment]
Hallo @codechefnisha-bp I deployed this internally and there something wrong with the readme
image
See those [7] and[8]? I think those are meant to be links?

**Pull request merged**